prog/heartbeat.rb

```diff
@@ -3,8 +3,14 @@
 require "excon"
 
 class Prog::Heartbeat < Prog::Base
+  CONNECTED_APPLICATION_QUERY = <<SQL
+SELECT count(DISTINCT application_name)
+FROM pg_stat_activity
+WHERE application_name ~ '^(bin/respirate|bin/monitor|.*/puma)$'
```

Comment:
Why do we need catch-all regex(.*) at the beginning for puma? Can that part change? If yes, would it cause any problem in our counting logic when we have multiple puma services running?

For example if we have 2 differently named puma and no monitor running, query would still return 3 and we wouldn't get any page.

Reply:
I think that's in the category of "when we get there" and at that point we should (or, we could elect to do this now....) set application_name ourselves.

puma has an arg0 (the default application name) of:

/app/vendor/bundle/ruby/3.2.0/bin/puma

As you can see...a lot of rather ungratifying text that could change for Reasons but not mean anything.

Reply:
In new version...things are more different still:

```sql
SELECT
  (regexp_matches(application_name, '/(puma|monitor|respirate)$'))[1]
FROM
  (SELECT DISTINCT application_name FROM pg_stat_activity) AS psa
ORDER BY 1
```